### PR TITLE
Add java truststore configmap

### LINKED ISSUE #52: Java Truststore from ConfigMap
I would like to be able to generate a java truststore from a configmap so I can utilize the "config.openshift.io/inject-trusted-cabundle" label.  This will enable me to consume the ca trusts that are configured at a cluster level for an environment and pass them on to Java applications and remove the need to create custom images for trust.

### PR DESCRIPTION
**Pull request by mathianasj**
closes #52

### CODE REVIEW

**Comment by mathianasj** [comment]
I need to add some readme updates, I should have marked this a draft

File: pkg/controller/cainjection/secret_controller.go [hunk 1]
```diff
@@ -58,6 +58,12 @@ func addSecretReconciler(mgr manager.Manager, r reconcile.Reconciler) error {
 			if oldSecretDest != newSecretDest {
 				return true
 			}
+
```
Comment (raffaelespazzoli) [inline pkg/controller/cainjection/secret_controller.go:61]: why did you have to change this file? from what I understand you were just adding a new controller...

Comment (mathianasj) [reply pkg/controller/cainjection/secret_controller.go:61]: It looks like stuff from when I thought I had to go that direction made it on the wrong branch let me clean it up

**Comment by raffaelespazzoli** [comment]
if we do this the way the key in the configmap representing the cabundle is going to be very OCP specific. This is good as a default, but can be find a way to let the user specify the key? Maybe an additional annotation?

**Comment by raffaelespazzoli** [comment]
@mathianasj did you see this comment:
```
if we do this the way the key in the configmap representing the cabundle is going to be very OCP specific. This is good as a default, but can be find a way to let the user specify the key? Maybe an additional annotation?
```
thoughts?

**Comment by mathianasj** [comment]
I am working on that now

**Comment by mathianasj** [comment]
> @mathianasj did you see this comment:
> 
> ```
> if we do this the way the key in the configmap representing the cabundle is going to be very OCP specific. This is good as a default, but can be find a way to let the user specify the key? Maybe an additional annotation?
> ```
> 
> thoughts?

I have implemented this, let me know if you think the annotation name is appropriate, also added a unit test to validate it

File: README.md [hunk 2]
```diff
@@ -71,6 +72,17 @@ A such annotated secret looks like the following:
 
 The default password for these keystores is `changeme`. The password can be changed by adding the following optional annotation: `cert-utils-operator.redhat-cop.io/java-keystore-password: <password>`. The alias of the certificate inside the keystore is `alias`.
 
+### ConfigMaps
```
Comment (raffaelespazzoli) [inline README.md:75]: can you update the docs to reflect the new annotation available?

Comment (mathianasj) [reply README.md:75]: added docs for annotations and defaults

**Pull request merged**